components/Catlist/index.js

```diff
@@ -1,7 +1,22 @@
 import Image from "next/image";
 import Link from "next/link";
+import BookmarkButton from "../Bookmarkbutton"; // Importieren der BookmarkButton-Komponente
+
+export default function List({ items, onBookmark }) {
+  const handleBookmark = (itemId, isBookmarked) => {
+    console.log(
+      `handleBookmark called with itemId: ${itemId}, isBookmarked: ${isBookmarked}`
+    ); // Füge diese Zeile hinzu
+    let favorites = JSON.parse(localStorage.getItem("favorites")) || [];
+    if (isBookmarked) {
+      favorites.push(itemId);
+    } else {
+      favorites = favorites.filter((favId) => favId !== itemId);
```

Comment:
if i get it right here, i would suggest sth. like:
`filteredFavorites = favorites.filter(....)`
for others to better understand whats inside variables, but also for yourself when some time has passed ✌🏼
(same on your other lists)